src/cargo/util/command_prelude.rs

```diff
@@ -1106,6 +1110,53 @@ pub fn get_registry_candidates() -> CargoResult<Vec<clap_complete::CompletionCan
     }
 }
 
+fn get_profile_candidates() -> CargoResult<Vec<clap_complete::CompletionCandidate>> {
+    let gctx = new_gctx_for_completions()?;
+    if let Ok(ws) = Workspace::new(&find_root_manifest_for_wd(gctx.cwd())?, &gctx) {
+        if let Ok(profiles) = Profiles::new(&ws, InternedString::new("dev")) {
+            let mut candidates = Vec::new();
+
+            for name in profiles.profile_names() {
+                if let Ok(profile_instance) = Profiles::new(&ws, name) {
+                    let base_profile = profile_instance.base_profile();
+
+                    let mut description = String::from(if base_profile.opt_level.as_str() == "0" {
+                        "unoptimized"
+                    } else {
+                        "optimized"
+                    });
+
+                    if base_profile.debuginfo.is_turned_on() {
+                        description.push_str(" + debuginfo");
+                    }
+
+                    if matches!(name.as_str(), "test" | "bench") {
+                        description.push_str(" with tests enabled");
+                    }
+
+                    candidates.push(
+                        clap_complete::CompletionCandidate::new(name.to_string())
+                            .help(Some(description.into())),
+                    );
+                }
+            }
+
+            return Ok(candidates);
+        }
+    }
+
+    Ok(vec![
```

Comment:
This is also `profile` in configuration file, though we might not want to go that far in completion code I guess?

https://github.com/rust-lang/cargo/blob/611b7c4ee40b63d0818040f7290a8e14424a9519/src/cargo/core/profiles.rs?plain=1#L1299-L1302

Reply:
In the main flow, I believe we do load `[profile]` from both config and workspace because we do `Profile::new().get_profile_names` which looked like the main code path for doing this.

This code path is for when that fails.  We could try to make it smarter and give people as much information as can successfully run  but that ends up duplicating a lot more of `Profiles`. My assumption is that it will be fine with the hard coded list.